### PR TITLE
Fix white flash in dark mode

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Fixes white flash when device is in dark mode.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/state/themes-mixin.ts [hunk 1]
```diff
@@ -88,9 +88,15 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
...
-      ).getPropertyValue("--app-header-background-color");
+      const computedStyles = getComputedStyle(document.documentElement);
+      const headerColor = computedStyles.getPropertyValue(
+        "--app-header-background-color"
+      );
+
+      document.documentElement.style.backgroundColor = computedStyles.getPropertyValue(
+        "--primary-background-color"
+      );
+
```
Comment (ludeeus) [inline src/state/themes-mixin.ts:99]: Should we add primary text color here as well?
Should fix #6814

**Review by ludeeus — APPROVED** [review]

File: src/state/themes-mixin.ts (hunk 1 above)
Comment (bramkragten) [reply src/state/themes-mixin.ts:99]: Help me, why is that happening now? The theme is loaded in onboarding right?

Comment (ludeeus) [reply src/state/themes-mixin.ts:99]: For me, I have seen it when I have set dark for localhost then spin up a fresh dev container.
I tested it, and adding primary-text-color as well, does indeed fix that issue.

Comment (bramkragten) [reply src/state/themes-mixin.ts:99]: I would like to know the source, can you reproduce it and check for errors? It is working fine here...

Comment (bramkragten) [reply src/state/themes-mixin.ts:99]: image

Comment (bramkragten) [reply src/state/themes-mixin.ts:99]: Let's move it to a new PR

**Comment by spacegaier** [comment]
@bramkragten For me I still have a white flash, but only in the area of the sidebar.

140B3728-9F3E-4969-A51C-33F6B715F884

File: src/html/index.html.template [hunk 2]
```diff
@@ -48,7 +48,7 @@
       }
       @media (prefers-color-scheme: dark) {
         html {
-          background-color: var(--primary-background-color, #111111);
+          background-color: #111111;
         }
         #ha-init-skeleton::before {
           background-color: #1c1c1c;
```
Comment (spacegaier) [inline src/html/index.html.template:54]: Is there a reason why this is not also `#111111`?

Comment (bramkragten) [reply src/html/index.html.template:54]: This is the header bar

**Pull request merged**